packages/lib/src/vis/shared/Annotation.tsx

```diff
@@ -16,11 +16,16 @@ function Annotation(props: Props) {
   const { x, y, scaleOnZoom, children, style, ...divProps } = props;
 
   const camera = useThree((state) => state.camera);
+  useFrameRendering();
 
   const { dataToWorld, worldToHtml } = useAxisSystemContext();
   const htmlPt = worldToHtml(dataToWorld(new Vector2(x, y)));
 
-  useFrameRendering();
+  if (scaleOnZoom && style?.transform) {
```

Comment:
We only want to forbid passing a `transform` if `scaleOnZoom` is `true`. It was a bit complicated to do this with types, so I went with a runtime assertion.